### PR TITLE
Crossplane ClusterRole expansion for composition

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

Using composition, Crossplane pod now provisions the managed resources directly and it needs the necessary ClusterRole to do that. There is already a `crossplane-admin` cluster role that when stacks are installed, the clusterrole to manage their CRs are aggregated to that persone. This PR expands the ClusterRole of Crossplane pod to include the ClusterRoles that `crossplane-admin` has so that the CRs of stacks can be part of composition.

In the end, we'll end up with Crossplane being able to deploy everything everywhere [as per the composition desig](https://github.com/crossplane/crossplane/pull/1163)n but this change only covers the case where managed resources are composed.

### How has this code been tested?

Manually tested with Azure stack.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation] and [examples].
- [x] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplane/crossplane/tree/master/design/one-pager-definition-of-done.md

### CODE REVIEW

File: cluster/charts/crossplane-types/templates/clusterrole.yaml [hunk 1]
```diff
@@ -7,6 +7,25 @@ metadata:
     chart: {{ template "chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+aggregationRule:
+  clusterRoleSelectors:
+  - matchLabels:
+      rbac.crossplane.io/aggregate-to-crossplane-admin: "true"
+  - matchLabels:
+      rbac.crossplane.io/aggregate-to-crossplane: "true"
```
Comment (negz) [inline cluster/charts/crossplane-types/templates/clusterrole.yaml:15]: Are you thinking that we could eventually have the stack manager add this label to newly installed roles, so we can differentiate between what `crossplane-admin` should grant and what `crossplane` should grant?

**Comment by negz** [comment]
> we are already trusting it to install the stacks which themselves have RBAC control over their CRDs

That's done by the stack-manager pod, right? I believe we decided to run stack-manager as a separate deployment in order to restrict the access the Crossplane service account gets. I wonder if that still makes sense? I will note that the stack-manager appears to run as `cluster-admin`, so even with these expanded Crossplane permissions the stack-manager likely has more.

File: cluster/charts/crossplane-types/templates/clusterrole.yaml (hunk 1 above)
Comment (muvaf) [reply cluster/charts/crossplane-types/templates/clusterrole.yaml:15]: Actually, I don't see a case where there is a ClusterRole that Crossplane creates with more permissions than Crossplane pod itself anymore (same treatment as stack-manager). So, the SA of Crossplane will have the most permissions regarding stacks and `crossplane-admin` would probably have the same level of permissions as it's _admin_. So I don't immediately see a case where we want to differentiate on stack level.

The differentiation exists in this file to some extent; aggregation `rbac.crossplane.io/aggregate-to-crossplane` is the minimal set that Crossplane needs without any stack. But as the stacks get installed, both `crossplane-admin` and `crossplane` end up having the same rules.

**Comment by hasheddan** [comment]
@negz I was meaning the trust a user was giving to crossplane in general terms, but it is a good point that it may not make sense to keep the core and stack manager separate any more

**Comment by jbw976** [comment]
I like this approach in that (I believe) it is essentially aggregating permissions to only types that the packages contain. It's not willy nilly taking cluster-admin permissions to `*`.  This keeps the scope of permissions in crossplane to a superset of all packages that have been installed.  

I had to refresh my memory on why stack-manager would be different and needs cluster-admin, which I'll leave here for convenience from the [security design doc](https://github.com/crossplane/crossplane/blob/master/design/one-pager-stacks-security-isolation.md#stack-manager):
> Currently, the Stack Manager that installs stacks into the cluster runs with a ClusterRoleBinding to the cluster-admin ClusterRole because it needs to create RBAC permissions for the arbitrary CRDs that are owned and depended on by arbitrary stacks. A service account cannot grant RBAC permissions that it doesn't already have itself, so to support the arbitrary permissions needs of all stacks, we are currently using the cluster-admin role.

**Comment by muvaf** [comment]
> This keeps the scope of permissions in crossplane to the set of all packages that have been installed.

@jbw976 Yeah I'm glad that we've been doing this for the clusterroles. It's a nice way to stay away from `*`.

**Pull request merged**